pkg/ingester/instance.go

```diff
@@ -120,6 +122,22 @@ func (i *instance) Push(ctx context.Context, req *logproto.PushRequest) error {
 	return appendErr
 }
 
+func (i *instance) getOrCreateStream(labels []client.LabelAdapter) (*stream, error) {
+	fp := client.FastFingerprint(labels)
+	stream, ok := i.streams[fp]
+	if !ok {
```

Comment:
Small nit: can we change this to an early return? e.g.,

```go 
if ok {
   return stream, nil
}
```

that way we don't have to have nested if statements.